internal/newusers/cache/db_test.go

```diff
@@ -429,23 +417,17 @@ func createDBFile(t *testing.T, src, destDir string) {
 	require.NoError(t, err, "Setup: should be able to write database file")
 }
 
-func requireNoDirtyFileInDir(t *testing.T, cacheDir string) {
-	t.Helper()
-
-	require.NoFileExists(t, filepath.Join(cacheDir, cachetests.DirtyFlagDbName), "Dirty flag should have been removed")
-}
-
 func TestMain(m *testing.M) {
```

Comment:
I would add tests (even if dummy) for the new exported API `Remove` and `Clear`. What happen if you remove twice, if you clear twice, and so on, there is room for table testing :)

Reply:
I thought about writing tests for them, but they felt a bit too forced and redundant, tbh. `Remove` is a wrapper around `os.Remove`, so no need to test that. `Clear` uses `Remove` and `openAndInitDb`, which is already quite covered due to `TestNew`. I could create dummy tests with a comment explaining why they are not necessary, but I'm not sure it's "worth it".

Reply:
I always think this is important that at least one test exercise the whole API to illustrate it. Basically, we should per package, check the coverage and see that we covered it all at an "unit/per package" level.

I’m sure you can demonstrate corner cases like clearing twice, and so on, that would be interested, so as said, there are content even for table testing :)